File: wallet/bindings/nodejs/examples/destroy-alias.js
/**
 * This example will destroy an alias
 */
const getUnlockedManager = require('./account-manager');

async function run() {
    try {
        const manager = await getUnlockedManager();

        const account = await manager.getAccount('0');

        await account.sync();
        console.log('Balance before destroying:\n', await account.getBalance());

        // Replace with an existing alias ID in your account
        let aliasId =
            '0x08e6210d29881310db2afde095e594f6f006fcdbd06e7a83b74bd2bdf3b5190d0e0200000000';

        await account.destroyAlias(aliasId);

        // Wait for transaction inclusion
        await new Promise(resolve => setTimeout(resolve, 5000));

        await account.sync();
        console.log('Balance after destroying:\n', await account.getBalance());
    } catch (error) {
        console.log('Error: ', error);
    }
    process.exit(0);
}

run();

File: wallet/bindings/python/examples/destroy_alias.py
from iota_wallet import IotaWallet
import time

# In this example we will destroy an alias

wallet = IotaWallet('./alice-database')

account = wallet.get_account('Alice')

# Sync account with the node
response = account.sync()
print(f'Balance before destroying:\n{response}')

wallet.set_stronghold_password("some_hopefully_secure_password")

# TODO: replace with your own values.
alias_id = "0x08429fe5864378ce70699fc2d22bb144cb86a3c4833d136e3b95c5dadfd6ba0cef0500000000"

# Send transaction.
account.destroy_alias(alias_id)

# Wait a few seconds for the transaction to get confirmed
time.sleep(7)

# Sync account with the node
response = account.sync()
print(f'Balance after destroying:\n{response}')
